usaspending_api/search/v2/views/spending_by_award.py

```diff
@@ -338,24 +395,22 @@ def construct_es_reponse(self, response) -> dict:
             row = self.append_recipient_hash_level(row)
             row.pop("parent_recipient_unique_id")
             results.append(row)
-        last_id = None
-        last_value = None
         if len(response) > 0:
-            last_id = response[len(response) - 1].to_dict().get("generated_unique_award_id")
-            last_value = (
-                response[len(response) - 1].to_dict().get("total_loan_value")
-                if set(self.filters["award_type_codes"]) <= set(loan_type_mapping)
-                else response[len(response) - 1].to_dict().get("total_obligation")
-            )
+            last_record_unique_id = int(response[len(response) - 1].to_dict().get("award_id"))
+            last_record_tie_break_value = float(response[len(response) - 1].to_dict().get("total_obligation"))
+            last_record_sort_value = response[len(response) - 1].to_dict().get(self.get_elastic_sort_by_fields()[0])
+        total = 0
+        if response != {}:
+            total = response.hits.total
         return {
             "limit": self.pagination["limit"],
             "results": results,
             "page_metadata": {
                 "page": self.pagination["page"],
-                "hasNext": response.hits.total - (self.pagination["page"] - 1) * self.pagination["limit"]
-                > self.pagination["limit"],
-                "last_id": last_id,
-                "last_value": last_value,
+                "hasNext": total - (self.pagination["page"] - 1) * self.pagination["limit"] > self.pagination["limit"],
+                "last_record_unique_id": last_record_unique_id,
+                "last_record_tie_break_value": last_record_sort_value,
+                "last_record_sort_value": last_record_tie_break_value,
```

Comment:
Did these get mixed up? `name: value`. The lefts don't match the rights. 
Or am I not seeing correctly how the the pagination logic is applied?

Reply:
YUP that's just a mistake